### DISCUSSION TITLE
reactive works while ref not work in pinia setup mode

### DISCUSSION THREAD

**Discussion by systemview2018** [vuejs/pinia#2839] [Help and Questions] [answered]
### Reproduction

1

### Steps to reproduce the bug

  const nickname = ref('')
  const menuList = reactive([])
 in vue file,use {nickname, menuList} = store.  
the nickname will be always the first value, but the menulist can be reactivity.

### Expected behavior

both can be reactivity

### Actual behavior

the nickname will be always the first value, but the menulist can be reactivity.

**Comment by posva** [comment, marked as answer]
This is because nickname is a primitive while menuList is an object. Object retains reactivity. Note you can directly use `store.anything` and that will retain reactivity. Or you can use `storeToRefs()` or `toRef()` too if you need to pass that data piece around.

**Comment by systemview2018** [comment]
I tested with const tttt = ref([]) and it does not work with const {tttt} = store.
It seems that only reactive works.

I am afraid of this beacause it may change in future, then I can not use reactive again.

**Reply by posva** [reply]
Don't restructure from the store if you want to keep reactivity. I recommend you to read the documentation for Pinia's Core Concepts and the [reactivity fundamentals of Vue](https://vuejs.org/guide/essentials/reactivity-fundamentals.html)
